### ISSUE TITLE
Customizable checkboxes and radio buttons

### ISSUE THREAD

**Issue by Suzanna-Manoj-Williams** [AllenFang/react-bootstrap-table#529]
Hi, 
I would like to customize the checkboxes and radiobuttons with custom class names and styles. Is there a way possible?

**Comment by AllenFang** [comment]
@Suzanna-Manoj-Williams, sorry, we didnt support this feature, check #497, I will make these component more customizable in ```v3.0.0```

**Comment by Suzanna-Manoj-Williams** [comment]
It will be helpful if the checkbox can be a combination of input and label for the input, like below:
--------------------------------------------------------------------------------------------------------------------------------------
<div className="checkbox">
	<input type="checkbox" id="testCB" value= "1"></input>
	<label htmlFor="testCB">Check!!</label>
</div>
--------------------------------------------------------------------------------------------------------------------------------------
So we can customize only the checkbox with label and not others.

**Comment by AllenFang** [comment]
@Suzanna-Manoj-Williams surely.

**Comment by hawker000** [comment]
I've found this feature is implemented in `v2.4.2`.
See an example [here](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/selection/custom-multi-select-table.js).

**Comment by AllenFang** [comment]
haha, yes, This feature has been supported, thanks @hawker000 your remind

**Comment by wenom64** [comment]
Hi,
it's possible add custom style for width (or change it) in lib\util.js@row63
30px is too narrow for custom checkboxes in my project.
```
if (isSelectRowDefined) {
  var style = {
    width: 30,
    minWidth: 30
  };
  if (!selectRow.hideSelectColumn) {
    selectRowHeader = _react2.default.createElement('col', { style: style, key: -1 });
  }
}
```

**Comment by AllenFang** [comment]
@wenom64, there's no any better solution, but you can use css to override it(There were someones did it)

**Comment by wenom64** [comment]
Too bad.. I don't like `!important`, but do I have a choise?! 😄 

My solution for now:
jsx
```jsx
<BootstrapTable
  tableHeaderClass="table-custom-checkboxes"
  tableBodyClass="table-custom-checkboxes"
>
```
stylesheet
```css
.table-custom-checkboxes > colgroup > col:first-child{
  width: 36px !important;
  min-width: 36px !important;
}
```

**Comment by AllenFang** [comment]
@wenom64, https://github.com/AllenFang/react-bootstrap-table/issues/979

**Comment by AllenFang** [comment]
@wenom64, released on `v2.10.0`, check [example](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/selection/selection-column-width-table.js#L24)

**Comment by wenom64** [comment]
Wow, U are the Best... I'm going to try it.
